### PR TITLE
Refactor RejectTransaction to CancelIOUTransaction

### PR DESCRIPTION
**Pull request by mountiny**
### Details
As part of the API unchained, we are refactoring our API calls so they are optimistic. Here the action of cancelling/declining a money request is made optimistic by creating the report action locally and then passing the new id to our API. 

Thanks to this, user can cancel money request offline and if everything was fine, this will be autocompleted once they get online.

In case of error, red brick road will lead him to the error showing him that cancellation failed.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/213615
$ https://github.com/Expensify/App/issues/8722

### Tests

This must be tested alongside with https://github.com/Expensify/Web-Expensify/pull/35088

**1. OFFLINE**
1. Go offline
2. Create a money request from User A to user B
3. Create another money request
4. As user A, go to the IOU details and **cancel** this IOU request
5. Make sure the request has been cancelled and the IOU preview has been updated (the message should be greyed out)
6. Go online and make sure that the cancel request is no longer greyed out
7. Refresh and switch chats there and back and make sure everything is fine
9. Create a money request from User A to user B
2. Create another money request
3. As user B, go offline
4. Go to the IOU details and **decline** this IOU request
5. Make sure the request has been declined and the IOU preview has been updated (the message should be greyed out)
6. Go online and make sure that the cancel request is no longer greyed out
7. Refresh and switch chats there and back and make sure everything is fine

**2. ONLINE**
1. Repeat offline steps, but skip any steps that ask you to go offline
2. Make sure that the requests are cancelled/declined

- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
Same as test steps.

- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/22219519/199350850-2261f561-8e0a-4979-9f69-53a2518ff670.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/22219519/199350896-50e322c4-af16-4620-ab28-2dc9845834ad.mov

#### Desktop

https://user-images.githubusercontent.com/22219519/199350920-4f45955d-4004-4d97-94f1-f11d48575083.mov

#### iOS

https://user-images.githubusercontent.com/22219519/199350932-c199dbe7-3a87-4acf-abba-24ec8d155652.mov

#### Android

https://user-images.githubusercontent.com/22219519/199350996-3136bbc7-a31b-478b-a052-4bfcc6acd3bd.mov

### CODE REVIEW

File: src/libs/actions/IOU.js [hunk 1]
```diff
@@ -201,41 +237,113 @@ function createIOUSplitGroup(params) {
...
+    // If we are cancelling the
+    if (iouReport.total === 0) {
+        chatReport.hasOutstandingIOU = false;
+    } else if (iouReport.total < 0) {
+        // The total sign has changed and hence we need to flip the manager and owner of the report.
+        const oldOwnerEmail = iouReport.ownerEmail;
+        iouReport.ownerEmail = iouReport.managerEmail;
+        iouReport.managerEmail = oldOwnerEmail;
+        iouReport.total = -iouReport.total;
+    }
```
Comment (mountiny) [inline src/libs/actions/IOU.js:291]: @luacmartins We will have to do this for Splits as well, there might be already existing request between the user A and some of the participant in which case the participant has requested less than is now being requested by user A so we will have to flip the sign and the manager and owner on that particular report.

Comment (luacmartins) [reply src/libs/actions/IOU.js:291]: Thanks! Created a method in my [PR](https://github.com/Expensify/App/blob/8f84ef50307c054127d1b86cd3714e9d4ba08522/src/libs/IOUUtils.js#L28-L58)

**Comment by mountiny** [comment]
Is this ready to be tested with the Web PR?

**Comment by luacmartins** [comment]
Ready for review!

**Comment by parasharrajat** [comment]
> Thanks to this, user can cancel money request online and if everything was fine, this will be autocompleted once they get online.

If I am not getting it wrong. 

> Thanks to this, user can cancel money request ~online~**offline** and if everything was fine, this will be autocompleted once they get online.

I have no access to. Can I still test it?
image

**Comment by luacmartins** [comment]
@parasharrajat we will have to wait for the API PR to hit staging. I'll ping you once that happens.

File: src/libs/ReportUtils.js [hunk 2]
```diff
@@ -701,6 +700,15 @@ function getIOUReportActionMessage(type, total, participants, comment, currency)
...
             iouMessage = `Split ${amount} with ${from}${comment && ` for ${comment}`}`;
             break;
+        case CONST.IOU.REPORT_ACTION_TYPE.CANCEL:
+            iouMessage = `Cancelled the ${amount} request${comment && ` for ${comment}`}`;
+            break;
+        case CONST.IOU.REPORT_ACTION_TYPE.DECLINE:
+            iouMessage = `Declined the ${amount} request${comment && ` for ${comment}`}`;
+            break;
+        case CONST.IOU.REPORT_ACTION_TYPE.PAY:
+            iouMessage = `Paid ${amount} to ${from}${comment && ` for ${comment}`}`;
```
Comment (mountiny) [inline src/libs/ReportUtils.js:710]: With this message, can you ever pay more people at once (now). It is always just 1:1 right?

**Review by mountiny — COMMENTED** [review]
@luacmartins Great work, just one comment also one issue from testing. 

Use a different currency. I am using VPN at the moment set to Czechia so my IOU comes up as CZK. When I create the IOU offline everythign works like a charm and adds/declines the values and updates them correctly.

When I go online, the IOU is however converted to USD so then it shows in USD, which is fine.

However, then:
1. Having IOU dispayed in total in USD
2. Go offline
3. Request more money in different currency (CZK in my case)
4. The value is added to the total which is however different currency
5. Hence the offline updated total is incorrect and stays so until you come online and the commands and processed and the correct dollar value is fetched.


Same happens if you:
1. Have a pending requests showing in dollars in the total
2. Go offline
3. Decline/Cancel the CZK or some other currency request
4. the value of that request is removed leaving incorrect total since we are doing math operation between dollars and different currencies.


I am not sure what is the best solution for this since we cannot really do the conversions locally

**Comment by mountiny** [comment]
Realizing this must be issue with all the optimistic actions in IOU flow :D CreateIOUTransaction must have the same issue same as CreateSplitBill

File: src/libs/ReportUtils.js (hunk 2 above)
Comment (luacmartins) [reply src/libs/ReportUtils.js:710]: I think we can only pay one report at a time and even when we split a bill we create multiple 1:1 iouReports, so I think this message is ok?

**Comment by luacmartins** [comment]
I think that we should tackle the currency issue you [mentioned](https://github.com/Expensify/App/pull/11720#pullrequestreview-1165596803) in another PR since that seems to be an issue with RequestMoney / SplitBill instead of CancelMoneyRequest.

**Comment by luacmartins** [comment]
@parasharrajat the Web-E PR is on staging. You can begin QA'ing this one!

**Comment by luacmartins** [comment]
Started a thread in Slack for the currency issue - https://expensify.slack.com/archives/C01GTK53T8Q/p1667416017878019

**Comment by parasharrajat** [comment]
I might not have time to review this today. Please reassign if needed. Thanks.

File: src/libs/ReportUtils.js (hunk 2 above)
Comment (mountiny) [reply src/libs/ReportUtils.js:710]: Ah the `from` is just tripping me up in this function, I maybe some comments showing what the values of `from` would be could be helpful to understand this method better.

**Review by mountiny — COMMENTED** [review]
I have asked if there would be any C+ willing to help with testing today.

@luacmartins there are some merge conflicts. Is there anything to implement here stemming from the discussion about different currencies?

**Comment by luacmartins** [comment]
@mountiny

>there are some merge conflicts

Resolved!

>Is there anything to implement here stemming from the discussion about different currencies?

I created a new issue for it https://github.com/Expensify/App/issues/12433, since the fix is actually in RequestMoney and eventually PayIOU and PayIOUWithWallet

>Ah the from is just tripping me up in this function, I maybe some comments showing what the values of from would be could be helpful to understand this method better.

Haha good point! I updated `from` to `who` which is the same name we use in PHP.

**Comment by luacmartins** [comment]
Hold removed!

**Comment by luacmartins** [comment]
Bumping @Luke9389 for review

File: src/libs/ReportUtils.js (hunk 2 above)
Comment (luacmartins) [reply src/libs/ReportUtils.js]: renamed the variable to `who` which is what we use in PHP

**Comment by Santhosh-Sellavel** [comment]
Reviewing this now!

File: src/pages/iouReportPropTypes.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import PropTypes from 'prop-types';
 
 export default PropTypes.shape({
     /** The report ID of the IOU */
-    reportID: PropTypes.string,
+    reportID: PropTypes.number,
```
Comment (Santhosh-Sellavel) [inline src/pages/iouReportPropTypes.js:5]: Confused I am seeing this somewhere, should this be a number or string?

 * @param {String} iouReportID

File: src/libs/actions/IOU.js [hunk 4]
```diff
@@ -535,41 +540,103 @@ function splitBillAndOpenReport(participants, currentUserLogin, amount, comment,
...
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+            value: {
+                [optimisticReportAction.sequenceNumber]: {
+                    ...optimisticReportAction,
+                    pendingAction: null,
+                    errors: {
+                        [DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage'),
+                    },
```
Comment (Santhosh-Sellavel) [inline src/libs/actions/IOU.js:623]: Need update error message, we are canceling here.

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/pages/iouReportPropTypes.js (hunk 3 above)
Comment (luacmartins) [reply src/pages/iouReportPropTypes.js:5]: We are transitioning to strings. So you'll find both right now. This particular one should be string though. Updated!

File: src/libs/actions/IOU.js (hunk 4 above)
Comment (luacmartins) [reply src/libs/actions/IOU.js]: Updated!

**Comment by luacmartins** [comment]
@Santhosh-Sellavel comments addressed!

**Comment by Santhosh-Sellavel** [comment]
@luacmartins  PR is Failing or Not?

Steps:
1. Go Offline
2. Created a money request (1000)
3. Created another money request (2000) 
4. Cancelled the last request (2000)

👍 Balance updated 1000
👎 Error message shown in chat (is this expected?)
👎  ❌  mark on the error did not work on the chat screen

5. So tapped on the IOU Balance message to view IOUDetails

👍 Error is displayed on the IOUDetails page also

6. Click ❌  mark on the here
👍  ❌  mark on the error worked on the IOUDetails page
👎  Balance didn't as we cancelled the cancel request

7. Go online
👎  Cancel requests succeed even after cancelling it

https://user-images.githubusercontent.com/85645967/199831485-0d882513-c2f3-4827-8191-04237b955953.mov

**Comment by luacmartins** [comment]
@Santhosh-Sellavel lol I left the code to test the error case in there. Sorry about that! Updated, please test again.

**Comment by Santhosh-Sellavel** [comment]
@luacmartins Thanks
One Small Issue  Look between 55 secs to 1:01 minute mark. Balance changes from 2000 to 4000 and back to 2000.

https://user-images.githubusercontent.com/85645967/199835794-e8d49110-4f25-4400-a6fd-50928f34384c.mov

**Comment by luacmartins** [comment]
@Santhosh-Sellavel thanks for raising that issue. I have a couple of observations:

1. Why is your cancel request failing while you are offline? These should always succeed offline.
2. I noticed that this bug happens because we changes `sequenceNumber` to `clientID` in [this PR](https://github.com/Expensify/App/pull/11538). I left a [comment](https://github.com/Expensify/App/pull/11538#discussion_r1013531761) there since this `ReportActions.deleteOptimisticReportAction` expects `sequenceNumber` as a 2nd param and not `clientID`.

**Review by mountiny — COMMENTED** [review]
The code looks good to me, again, I cannot approve given I have also pushed to this branch so gonna ask someone else for this. @Santhosh-Sellavel seems like you have tested thoroughly and the bug found will be fixed in different PR.

Would you be able to test it and leave the reviewer checklist?

@luacmartins there are some merge conflicts, I will resolve them

**Comment by mountiny** [comment]
Updated

File: src/components/ReportTransaction.js [hunk 5]
```diff
@@ -27,17 +27,10 @@ const propTypes = {
     /** Can this transaction be rejected? */
     canBeRejected: PropTypes.bool,
 
-    /** Text label for the reject transaction button */
-    rejectButtonLabelText: PropTypes.string.isRequired,
+    /** Type of the reject transaction button */
+    rejectButtonType: PropTypes.oneOf(['decline', 'cancel']).isRequired,
```
Comment (Gonals) [inline src/components/ReportTransaction.js:31]: NAB: Should we use constants here?

File: src/libs/actions/IOU.js [hunk 6]
```diff
@@ -210,7 +215,7 @@ function requestMoney(report, amount, currency, recipientEmail, participant, com
         comment,
         iouReportID: iouReport.reportID,
         chatReportID: chatReport.reportID,
-        transactionID: NumberUtils.rand64(),
+        transactionID: optimisticReportAction.originalMessage.IOUTransactionID,
```
Comment (Gonals) [inline src/libs/actions/IOU.js:218]: Ah, nice!

**Review by Gonals — DISMISSED** [review]

**Comment by Santhosh-Sellavel** [comment]
@mountiny or @luacmartins Release conflicts, please!

**Comment by luacmartins** [comment]
@Santhosh-Sellavel updated!

**Comment by mountiny** [comment]
LGTM! We should get someone internal to approve too

**Review by Gonals — APPROVED** [review]

**Review by Luke9389 — APPROVED** [review]

**Comment by Santhosh-Sellavel** [comment]
Testing this now!

**Comment by Santhosh-Sellavel** [comment]
<details>
<summary><h4>Screen Recordings</h4></summary>

#### WEB & DESKTOP 
##### OFFLINE/ONLINE CANCEL
https://user-images.githubusercontent.com/85645967/200063620-7b2b6578-4b6f-43d7-b843-0c5300627573.mov

##### OFFLINE/ONLINE DECLINE
https://user-images.githubusercontent.com/85645967/200063638-38f0d160-460b-4372-929c-da81846a84e5.mov

##### ONLINE DECLINE & CANCEL
https://user-images.githubusercontent.com/85645967/200062000-53749a79-5222-4a7b-bc0b-e18b8f6bf4b0.mov

</details>

**Comment by Santhosh-Sellavel** [comment]
details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

**Comment by Santhosh-Sellavel** [comment]
Checklist: https://github.com/Expensify/App/pull/11720#issuecomment-1304268561
Recordings: https://github.com/Expensify/App/pull/11720#issuecomment-1304114329

Only one unanswered question [here](https://github.com/Expensify/App/pull/11720#issuecomment-1302675155). What about toggling balance?
As the above issue might be unrelated to this as it could occur during the creation flow also, so approving this one!

Looks good!
C+ Reviewed
🎀 👀 🎀

**Review by Santhosh-Sellavel — APPROVED** [review]
Checklist: https://github.com/Expensify/App/pull/11720#issuecomment-1304268561
Recordings: https://github.com/Expensify/App/pull/11720#issuecomment-1304114329

Only one unanswered question [here](https://github.com/Expensify/App/pull/11720#issuecomment-1302675155). What about toggling balance?
As the above issue might be unrelated to this as it could occur during the creation flow also, so approving this one!

Looks good!
C+ Reviewed
🎀 👀 🎀

**Comment by luacmartins** [comment]
@Santhosh-Sellavel I address that comment here https://github.com/Expensify/App/pull/11720#issuecomment-1302858094. This is unrelated to this PR and I'll create another issue for this.

**Comment by luacmartins** [comment]
Merging this PR since we have 3 approvals!

**Comment by luacmartins** [comment]
Thanks everyone for the reviews and tests!

**Comment by luacmartins** [comment]
Tests passed. Removing `Emergency` label.

**Pull request merged**